lib/opensusebasetest.pm

```diff
@@ -981,29 +982,25 @@ base method using C<$self-E<gt>SUPER::post_fail_hook;> at the end.
 
 sub post_fail_hook {
     my ($self) = @_;
-    return if is_serial_terminal();    # unless VIRTIO_CONSOLE=0 nothing below make sense
-
-    show_tasks_in_blocked_state;
 
     return if (get_var('NOLOGS'));
 
     # Upload basic health check log
-    select_log_console;
+    select_serial_terminal();
```

Comment:
i really think that you have to check whether serial_terminal is already used to avoid calling this unnecessarily. Many tests are already calling ` select_serial_terminal`. Although `select_console` does some checks, I think it would be better if you prevent to run the whole stuck when serial terminal is already set. WDYT?

Reply:
You can use `select_serial_terminal` multiple times.